xunit-autolabeler-v2/ast_parser/lib/constants.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import re
+
+
+IGNORED_METHOD_NAMES = (
+    'run_command',
+    'parse_command_line_args',
+    'main'
+)
+
+REGION_TAG_GREP_ARGS = (
+    'grep', '-hr', 'START', '.',
+
+    # Language-specific arguments
+    '--include=*.py', '--exclude=*/appengine/**/lib/**/*.py',
+    '--include=*.js', '--exclude=*/node_modules/*',
```

Comment:
Practically we're fine now, but I still think the design is wrong. Because we're always adding these arguments, we are skipping `node_modules` directory in Python repo.


I also think targeting only python files is not perfect because sometimes we have region tags in non-python files ([example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/1c9244b66ab90290054e8258aea718abc12eb944/functions/billing/requirements.txt#L2)).

Thoughts?

Reply:
I can add `requirements.txt`, `requirements-dev.txt`, and `package.json` here.

(I'm reluctant to get less specific than that, as it may pick up cruft elsewhere.)

Reply:
Maybe you can find all the region tags in python-docs-samples, and node samples repo and see what kind of file names have region tags?

Reply:
Sorry, but I'm thinking more about this.

You're shelling out for running grep to get the region tags.
Can we instead open the file and scan with regex?

I'm asking because 1) shelling out might fail 2) we can reuse the `_getFiles` (or `_get_files`) to get the filename so that we can change the include/exclude rules per language 3) you can directly extract the region tag name from the matched object

Thoughts?